src/dolphin/install/installation.ts

```diff
@@ -176,11 +176,16 @@ export class DolphinInstallation {
   }
 
   private async _isOutOfDate(latestVersion: string): Promise<boolean> {
+    const dolphinVersion = await this.getDolphinVersion();
+    return !dolphinVersion || lt(dolphinVersion, latestVersion);
+  }
+
+  public async getDolphinVersion(): Promise<string | null> {
     const dolphinPath = await this.findDolphinExecutable();
     const dolphinVersionOut = spawnSync(dolphinPath, ["--version"]).stdout.toString();
     const match = dolphinVersionOut.match(semverRegex);
     const dolphinVersion = match?.[0] ?? "";
-    return lt(dolphinVersion, latestVersion);
+    return dolphinVersion || null;
```

Comment:
```suggestion
    return dolphinVersion;
```
You should change the previous line to `match?.[0] ?? null` instead of changing the return.